pkg/kv/kvserver/store.go

```diff
@@ -3011,6 +3012,9 @@ func (s *Store) updateReplicationGauges(ctx context.Context) error {
 		if metrics.Quiescent {
 			quiescentCount++
 		}
+
+		uninitializedCount = int64(len(s.mu.uninitReplicas))
```

Comment:
Weird, github reviews don't show where this map is being updated, but I could see it locally. In any case, some notes from when we looked at this offline:

- The per-replica visitor above (`newStoreReplicaVisitor`) also iterates over uninitialized replicas. Instead of tracking+maintaining the set of uninitialized replicas in a separate map (probably more error prone), using the existing replica iteration + checking if the current replica is initialized is I think sufficient.
- We'd want a test. We looked at another test to steal from where we could block the reception of a snapshot on a second node, ensure that our uninitialized count is as we'd expect, unblock the replica initialization, and ensure it's back to 0.